pkg/infra/baseAgent.go

```diff
@@ -262,75 +265,58 @@ func (a *Base) UpdateHPChange(change int) {
 	}
 }
 
-func (a *Base) utility() float64 {
-	// TODO: Improve utility calculation
-	// Currently just use ratio between food taken and food seen
-	// Somehow also factor
-	// Ideas: piecewise/bounded function with log, concave quadratic
+// Updates the utility of a given agent:
+func (a *Base) updateUtility(foodRequested, foodTaken food.FoodType) {
+	// We take N agents i ∈{1,...,N}that perform the following actions in each iterated round t ∈{1,...,∞}:
+	// 1.) Determines the resources it has available, gi ∈[0,1]
+	// 2.) Determines its need for resources, qi ∈[0,1]
+	// 3.) Makes a provision of resources, pi ∈[0,1]
+	// 4.) Makes a demand for resources, di ∈[0,1]
+	// 5.) Receives an allocation of resources, ri ∈[0,1]
+	// 6.) Makes an appropriation of resources, r′i ∈[0,1]
```

Comment:
I mean put each line above the stage where it is relevant, so move the comment 1.) to above the line where `resAvail` is declared, and so on.

Don't need comment 5.